library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -624,91 +632,124 @@ void vtkF3DRenderer::ConfigureHDRI()
     }
 
     this->GetEnvMapPrefiltered()->HalfPrecisionOff();
+    needHDRIComputation = !lutCacheExists || !shCacheExists || !specCacheExists;
 #endif
 
-    // HDRI OpenGL
     this->UseImageBasedLightingOn();
-    this->SetEnvironmentTexture(hdriTexture);
 
-#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 2, 20221220)
-    // Create LUT cache file
-    if (!lutCacheExists)
+    if (needHDRIComputation || this->HDRISkyboxVisible)
     {
-      vtkPBRLUTTexture* lut = this->GetEnvMapLookupTable();
-      if (!lut->GetTextureObject())
+      // The actual env texture is needed, read it
+      textureReader->SetFileName(this->HDRIFile.c_str());
+      textureReader->Update();
+
+      hdriTexture->SetColorModeToDirectScalars();
+      hdriTexture->MipmapOn();
+      hdriTexture->InterpolateOn();
+      hdriTexture->SetInputConnection(textureReader->GetOutputPort());
+
+      // 8-bit textures are usually gamma-corrected
+      if (textureReader->GetOutput() &&
+        textureReader->GetOutput()->GetScalarType() == VTK_UNSIGNED_CHAR)
       {
-        this->Render();
-        lut = this->GetEnvMapLookupTable();
+        hdriTexture->UseSRGBColorSpaceOn();
       }
-      assert(lut->GetTextureObject());
-
-      vtkSmartPointer<vtkImageData> img = ::SaveTextureToImage(
-        lut->GetTextureObject(), GL_TEXTURE_2D, 0, lut->GetLUTSize(), VTK_UNSIGNED_SHORT);
-      assert(img);
-
-      vtkNew<vtkXMLImageDataWriter> writer;
-      writer->SetFileName(lutCachePath.c_str());
-      writer->SetInputData(img);
-      writer->Write();
     }
-
-    // Create spherical harmonics cache file
-    if (!shCacheExists)
+    else
     {
-      vtkNew<vtkTable> table;
-      table->AddColumn(this->SphericalHarmonics);
+      // TODO Improve vtkOpenGLRenderer to support not having a texture
+      vtkNew<vtkImageData> img;
+      hdriTexture->SetInputData(img);
 
-      vtkNew<vtkXMLTableWriter> writer;
-      writer->SetInputData(table);
-      writer->SetFileName(shCachePath.c_str());
-      writer->Write();
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 2, 20221220)
```

Comment:
is it required?

Reply:
SphericalHarmonics is only defined if  VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 2, 20221220)

Reply:
Are you sure? I implemented it way before December 2022 :)

Reply:
yes but without cache support (which seems to require 20221220), what is the point of calling Modified ?

Reply:
You are not going through this code path without cache support because `needHDRIComputation` is true.

Reply:
I need to check something because without it 9.0.0 would not build. It does not change anything to check an earlier date than 20221220, at least this way it is coherant that this call is about the cache mechanism

Reply:
Right. Just in case, here's the MR: https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6708